src/xtgeo/well/_well_oper.py

```diff
@@ -278,19 +279,23 @@ def get_gridproperties(self, gridprops, grid=("ICELL", "JCELL", "KCELL"), prop_i
     The routine will make grid_coordinates from grid with make_ijk_from_grid(), or reuse
     existing vectors if grid is a tuple (much faster).
     """
-    if not isinstance(gridprops, (xtgeo.GridProperty, xtgeo.GridProperties)):
+    from xtgeo.grid3d.grid import Grid
+    from xtgeo.grid3d.grid_properties import GridProperties
+    from xtgeo.grid3d.grid_property import GridProperty
```

Comment:
Another ugliness on my part. To me this indicates that this function is probably live somewhere closer to `xtgeo.grid3d` or somewhere in-between